cachelib/shm/FileShmSegment.cpp

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cachelib/shm/FileShmSegment.h"
+
+#include <fcntl.h>
+#include <folly/logging/xlog.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "cachelib/common/Utils.h"
+
+namespace facebook {
+namespace cachelib {
+
+constexpr static mode_t kRWMode = 0666;
+typedef struct stat stat_t;
+
+namespace detail {
+
+// TODO(SHM_FILE): move those *Impl functions to common file, there are copied
+// from PosixShmSegment.cpp
+static int openImpl(const char* name, int flags) {
+  const int fd = open(name, flags);
+
+  if (fd != -1) {
+    return fd;
```

Comment:
I believe the preference should be to have a single return statement for each function. It improves readability and simplifies maintainability.